src/test/regress/sql/logical_replication.sql

```diff
@@ -17,17 +17,23 @@ INSERT INTO dist SELECT generate_series(1, 100);
 
 SELECT 1 from citus_add_node('localhost', :master_port, groupId := 0);
 
+-- Create a publiction and subscription (including replication slot) manually.
+-- This allows us to test the cleanup logic at the start of the shard move.
 \c - - - :worker_1_port
 SET search_path TO logical_replication;
 CREATE PUBLICATION citus_shard_move_publication_:postgres_oid FOR TABLE dist_6830000;
 
 \c - - - :master_port
 SET search_path TO logical_replication;
-\set connection_string '\'user=postgres host=localhost port=' :worker_1_port '\''
+CREATE TABLE dist_6830000(
+    id bigserial PRIMARY KEY
+);
+\set connection_string '\'user=postgres host=localhost port=' :worker_1_port ' dbname=regression\''
```

Comment:
the dbname=regression was the initial fix, but when fixing that other errors occured, for which I did the rest of the changes.

Reply:
good catch, thanks!